docs/architecture/application/forms-frontend.md

```diff
@@ -0,0 +1,231 @@
+
+-   A `POST` request to `/processes/<workflow_name>` with an empty payload
+-   The backend determines what input values are missing and sends a response with http status code `510` and a payload containing a [JSON6Schema definition][2] describing the form to display. See [Example JSON6Schema response](#example-json6schema-response)
+-   The frontend uses the [Uniforms library][1] to parse the JSON response into a form to display
+-   The [AutofieldLoader function][3] is called for each of the form.properties in the JSON response. This functions uses the properties `type` and `format` to determine what kind of field will be displayed.
+
+```
+
+In the example json response below one of the properties is
+...
+            "customer_id": {
+                "default": "c9b5e717-0b11-e511-80d0-005056956c1a",
+                "format": "customerId",
+                "title": "Customer Id",
+                "type": "string",
+                "uniforms": {
+                    "disabled": true,
+                    "value": "c9b5e717-0b11-e511-80d0-005056956c1a"
+                }
+            }
+...
+
+In the autoFieldFunction this maps to a CustomerField.
+
+export function autoFieldFunction(
+    props,
+    uniforms,
+) {
+    const { allowedValues, checkboxes, fieldType, field } = props;
+    const { format } = field;
+
+    switch (fieldType) {
+    ...
+        case String:
+            switch (format) {
+                ....
+                case 'customerId':
+                    return CustomerField;
+                ...
+            }
+    ...
+
+The CustomerField is a React component that is provided by the Orchestrator Component Library.
+It's passed the complete property object so it can use them to adjust it's behavior.
+```
+
+-   A `POST` with the form values is made to the same `/processes/<workflow_name>` endpoint
+-   The response status code can be:
+    -   `400: Form invalid` Invalid values have been detected because the validators the backend runs have failed. An error message is shown.
+    -   `510: FormNotComplete` There is another step. This response contains another json response containing a form.
+    -   `201: Created` The workflow was initiated successfully. The response contains a workflow id and the user is redirected to the workflow detail page
+
+**Note**. For forms that have multiple steps the user input for each step is accumulated in local frontend state and posted to `/processes/<workflowname>` on each step. The endpoint will receive all available user inputs on each step and determine what other user input it till needs or if it's ready to start the workflow.
+
+**Note 2** The Orchestrator Component library contains fields that are marked as deprecated and live in a folder named `deprectated`. These contain field types that are very specific to workflows that are in use by SURF. There are plans to remove these from the general purpose components library.
```

Comment:
`deprectated` or `deprecated`?

Reply:
I'm pretty sure it's ``deprecated`` https://www.google.com/search?q=deprectated

Reply:
Fixed